src/test/scala/ShoppingCartSpec.scala

```diff
@@ -0,0 +1,152 @@
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.must.Matchers
+
+case class Price private (value: BigDecimal) {
+  def taxPayable(implicit taxRate: TaxRate): Price = Price(value * taxRate.multiplier)
```

Comment:
why is it an implicit? I guess in my head it's a context parameter, it definitely doesn't need to be an implicit it could just be passed without that - but writing implicit let me avoid some typing in this example where I guess in my head reducing boilerplate might make it clearer to explain 🤷 

I think probably having it get passed to the taxPayable method of the shopping cart probably isn't correct when I think about how tax is done more-  probably tax rate should be associated with a priced item - or a category of item that a priced item falls under, and then you could have on a priced item the taxpayable / totalpayable methods too and the shopping cart just delegates to those to work it out

but I don't know if what you wanted was simpler than that - or if you wanted something at a different level of abstraction based on feedback of it being unfinished, like a service that can handle some operations like a command to `AddProductToCart(CartId, ProductName, Quantity)` or a query to `GetCartSummary(CartId)`